Scripts/generate-test-config.sh

```diff
@@ -1,14 +1,269 @@
 #!/bin/bash
 
-DESTINATION=$1
+# vim: set ts=2 sw=2 sts=2 et:
 
-if [[ -z $DESTINATION ]] ; then
-  echo "Destination not specified" >&2
+##################################################
+# SCRIPT METADATA
+
+SCRIPT_NAME=$(basename $0)
+SUMMARY="Generates a testconfiguration.json file for use in Amplify and AWS mobile SDK integration tests"
+SYNOPSIS="${SCRIPT_NAME} <android|ios> [options]"
+
+
+##################################################
+# DEFAULTS & DECLARATIONS
+##################################################
+LOG_LEVEL_INFO=0
+LOG_LEVEL_DEBUG=1
+LOG_LEVEL_TRACE=2
+LOG_LEVEL=0
+
+
+##################################################
+# FUNCTIONS
+##################################################
+
+# Logging. Note that all logging output goes to STDERR
+
+function log {
+  msg="$(date) $*"
+  echo $msg >&2
+}
+
+function log_debug {
+  if [[ $LOG_LEVEL -lt LOG_LEVEL_DEBUG ]] ; then
+    return
+  fi
+
+  log "[D]:: $*"
+}
+
+# Always logged
+function log_error {
+  log "[E]:: $*"
+}
+
+# Always logged, except in quiet mode
+function log_info {
+  if [[ $LOG_LEVEL -lt LOG_LEVEL_INFO ]] ; then
+    return
+  fi
+
+  log "[I]:: $*"
+}
+
+function log_trace {
+  if [[ $LOG_LEVEL -lt LOG_LEVEL_TRACE ]] ; then
+    return
+  fi
+
+  log "[T]:: $*"
+}
+
+function die {
+  log_error $*
   exit 1
+}
+
+# Usage & help
+
+function print_usage {
+  echo "usage: ${SYNOPSIS}"
+}
+
+function print_help {
+  usage=$(print_usage)
+  cat <<EOF
```

Comment:
It won't be much of an improvement, but the `<<-EOF` will at least allow you to pad all of this content into the document with a `\t`, so that the function block is more clear.

Reply:
That's another one I considered but rejected, because the non-indented form makes it easier to see the width of the text as it will be output at runtime. If you're aware of a good way of transparently handling text wrapping I'll reconsider this, but since I'm currently manually handling line breaks in the heredoc, I wanted to know the "real" width without having to mentally subtract the indention space.